### PR TITLE
Fix fill outline rendering

### PR DESCRIPTION
**Pull request by mpulkki-mapbox**
Anti-aliasing implementation used in fill outline shader is erroneously comparing coordinates with different interpolation modes. `gl_FragCoord` is interpolated linearly in screen space whereas varyings are using perspective correct interpolation. This results in lines disappearing when map pitch is set to > 0 as seen in following animations:

bug_0
bug_1

This can be fixed by either using `noperspective` qualifier with varying variables or doing it by hand in the shader code. The latter approach is used in this PR as `noperspective` is not supported in webgl 1 & 2.

fix

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Fixed disappearing fill outlines when map pitch angle was greater than 0</changelog>`

### CODE REVIEW

**Review by karimnaaji — COMMENTED** [review]
Looks good, nice find! One nit on the changelog entry before we can merge.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### 🐞 Bug fixes
 
 - Fixed a font glyphs performance regression on Firefox. ([10363](https://github.com/mapbox/mapbox-gl-js/pull/10363))
+- Fixed disappearing fill outlines when map pitch angle was greater than 0. ([10456](https://github.com/mapbox/mapbox-gl-js/pull/10456))
```
[the remark this answers is not in the document]
Comment (mpulkki-mapbox) [reply CHANGELOG.md:6]: That's great! I'll move the changelog entry to the PR description :+1:

**Comment by mpulkki-mapbox** [comment]
Render test `text-pitch-alignment/viewport-text-depthtest` has started failing as part of this pull request. After further investigation the test seems to be already broken but subtle changes introduced in this PR if enough to throw the image diff over the threshold value.

Actual and expected images using `origin/main` branch.
actual
expected

**Review by karimnaaji — APPROVED** [review]
The change looks good, for https://github.com/mapbox/mapbox-gl-js/pull/10456#issuecomment-797449014, I think you can rebaseline the test for this PR and open a separate ticket for follow up, as it's not a regression introduced here.

**Pull request merged**